internal/objectstore/base.go

```diff
@@ -301,6 +301,10 @@ func (w *baseObjectStore) prune(ctx context.Context, list pruneListFunc, delete
 	return nil
 }
 
+func selectHash(m objectstore.Metadata) string {
+	return m.Hash512_384
+}
```

Comment:
Without context this could be a bit of a wtf. Perhaps rename to something like `getObjectFilename`, or include a docstring explaining what's going on here?

Reply:
This would hopefully make callers to `selectHash` a bit clearer

e.g. we have `fileObjectStore.filePath` which does something similar.

Reply:
More broadly, I'm a bit concerned that the 'knowledge of` the fact we store blobs at their hash_384 has leaked into a few places across the implementations of the object stores. It would be nice if this knowledge could be encapsulated into a single place, like here.

However, that will take more work than it's probs worth atm

Small improvement in that direction could be for `filePath` on the two object stores to accept the metadata, and to pick the specific hash out of there. Then we could theoretically drop this fella?

I'm not asking you do that now necessarily, just something to consider